src/components/Home/Today1Hourly/Today1Hourly.tsx

```diff
@@ -0,0 +1,165 @@
+import Card from '../../UI/Card/Card';
+import ErrorComponent from '../../UI/ErrorComponent/ErrorComponent';
+import Loading from '../../UI/Loading/Loading';
+import { ENDPOINTS, getData } from './../../../helpers/api';
+import Title from './../../UI/Title/Title';
+import styles from './Today1Hourly.module.scss';
+
+interface Today1HourlyProps {
+  location: string;
+  tempUnit: TempUnits;
+}
+interface Today1HourlyState {
+  data: HourlyData;
+  loading: boolean;
+  error: string;
+}
+
+class Today1Hourly extends React.Component<Today1HourlyProps, Today1HourlyState> {
+  constructor(props: Today1HourlyProps) {
+    super(props);
+    this.state = {
+      data: { forecast: [] },
+      loading: false,
+      error: ''
+    };
+  }
+
+  getHourlyData(): void {
+    getData(ENDPOINTS.hourly, this.props.location).then(
+      (data: HourlyData) => {
+        this.setState(() => ({ loading: true, data }));
+      },
+      (error: Error | AxiosError) => {
+        this.setState(() => ({ error: error.message }));
+      }
+    );
+  }
+
+  componentDidMount() {
+    this.getHourlyData();
+  }
+  componentDidUpdate(prevProps: { location: string }) {
+    if (prevProps.location !== this.props.location) {
+      this.getHourlyData();
+    }
+  }
+
+  render() {
+    const { data, loading, error } = this.state;
+    const { tempUnit } = this.props;
+    return (
+      <Card>
+        <div id="hourly">
+          <Title title="Hourly forecast" />
+          {data.forecast.length > 0 ? (
+            <>
+              <div
+                className={`${styles.container} ${
+                  this.context.theme === Theme.DARK && styles[Theme.DARK]
+                }`}
+              >
+                <svg height={'400'} width={'1000'} className={styles.svgBackground}>
+                  <text x={'0'} y={'15'} className={styles.tempScale}>
+                    {tempUnit === TempUnits.FAHRENHEIT
+                      ? `${convertToFahrenheit(40)} °F`
+                      : `${40} °C`}
+                  </text>
+                  <text x={'0'} y={'100'} className={styles.tempScale}>
+                    {tempUnit === TempUnits.FAHRENHEIT
+                      ? `${convertToFahrenheit(30)} °F`
+                      : `${30} °C`}
+                  </text>
+                  <text x={'0'} y={'200'} className={styles.tempScale}>
+                    {tempUnit === TempUnits.FAHRENHEIT
+                      ? `${convertToFahrenheit(20)} °F`
+                      : `${20} °C`}
+                  </text>
+                  <text x={'0'} y={'300'} className={styles.tempScale}>
+                    {tempUnit === TempUnits.FAHRENHEIT
+                      ? `${convertToFahrenheit(10)} °F`
+                      : `${10} °C`}
+                  </text>
+                  <text x={'0'} y={'400'} className={styles.tempScale}>
+                    {tempUnit === TempUnits.FAHRENHEIT ? `${convertToFahrenheit(0)} °F` : `${0} °C`}
+                  </text>
```

Comment:
suggestion: you can create function to render <text> with provided props like this:

```
const fn = (coords, value, className = styles.tempScale) => (
  <text x={coords.x} y={coords.y} className={className}>
    {tempUnit === TempUnits.FAHRENHEIT
      ? `${convertToFahrenheit(value)} °F`
      : `${value} °C`}
  </text>
)
```